tests/test_tool_versioning.py

```diff
@@ -5,33 +5,34 @@
 from tests import TestAnalysisDriver
 
 
-def fake_check_version(self, toolname, executable, exp_version, version_cmd):
+def fake_check_version(self, toolname, executable, exp_version, version_cmd, dependency):
     return executable.split('_')[1] == str(exp_version)
 
 
 class TestToolset(TestAnalysisDriver):
     def setUp(self):
         self.toolset = Toolset()
-        self.toolset.versioning_cfg = Configuration(join(TestAnalysisDriver.assets_path, 'tool_versioning.yaml'))
+        self.toolset.versioning_cfg = Configuration(join(self.assets_path, 'tool_versioning.yaml'))
         self.toolset.select_type('non_human_sample_processing')
 
     @patch('analysis_driver.tool_versioning.Toolset.check_version', new=fake_check_version)
     def test_tool_paths(self):
         self.toolset.select_version(0)
         assert self.toolset['bwa'] == 'path/to/bwa_1.0'
-        assert self.toolset['gatk'] == 'path/to/GenomeAnalysisTK.jar_3'
+        assert self.toolset['gatk'] == 'path/to/gatk_3'
         assert self.toolset['fastqc'] == 'path/to/fastqc_v0.11.5'
         assert self.toolset['samtools'] == 'path/to/samtools_1.3.1'
         assert self.toolset['qsub'] == '/bin/sh'
         assert 'qsub' in self.toolset.unversioned_tools
-        assert self.toolset.tool_versions == {'bwa': 1.0, 'fastqc': 'v0.11.5', 'gatk': 3, 'samtools': '1.3.1'}
+        assert self.toolset.tool_versions == {'bwa': 1.0, 'fastqc': 'v0.11.5', 'gatk': 3, 'samtools': '1.3.1', 'java': 8}
 
         self.toolset.select_version(1)
         assert self.toolset['bwa'] == 'path/to/bwa_1.1'
-        assert self.toolset['gatk'] == 'path/to/GenomeAnalysisTK.jar_4'
+        assert self.toolset['gatk'] == 'path/to/gatk_4'
         assert self.toolset['fastqc'] == 'path/to/fastqc_v0.11.5'
+        assert self.toolset['java'] == 'path/to/java_8'
         assert 'fastqc' in self.toolset.unversioned_tools
-        assert self.toolset.tool_versions == {'bwa': 1.1, 'gatk': 4, 'samtools': '1.3.1'}
+        assert self.toolset.tool_versions == {'bwa': 1.1, 'gatk': 4, 'samtools': '1.3.1', 'java': 8}
 
```

Comment:
It might be worth adding picard here too?

Reply:
Not really - it doesn't matter what the tools/toolsets are called here, we're only testing the inheritance of versions/commands.

Reply:
Right, for a second I thought you were testing the contents of the file.